app/characteristics/potion.py

```diff
@@ -0,0 +1,10 @@
+def amply_potion(knight):
+    if knight["potion"] is not None:
+        if "power" in knight["potion"]["effect"]:
+            knight["power"] += knight["potion"]["effect"]["power"]
+
+        if "protection" in knight["potion"]["effect"]:
+            knight["protection"] += knight["potion"]["effect"]["protection"]
+
+        if "hp" in knight["potion"]["effect"]:
+            knight["hp"] += knight["potion"]["effect"]["hp"]
```

Comment:
These code duplications can be also fixed, using `stats = ("protection", "power", "hp") `and looping on `stats`